File: selfstudy/golang/transactions_filterhigh.go
/*
task 235:
Transactions - FilterHigh
You need to return only the values greater than 50 from transaction amounts in dollars
*/

package main

import (
	"fmt"
	"reflect"
)

func Above_fiftyTransactions(amounts []int) []int {
	var res []int
	for _, v := range amounts {
		if v > 50 { res = append(res, v) }
	}
	return res
}

func main() {
	tests := []struct {
		input []int
		expected []int
	}{
		{[]int{10, 20, 30}, Above_fiftyTransactions([]int{10, 20, 30})},
		{[]int{50, 60, 70}, Above_fiftyTransactions([]int{50, 60, 70})},
		{[]int{1, 2, 3}, Above_fiftyTransactions([]int{1, 2, 3})},
		{[]int{100, 200}, Above_fiftyTransactions([]int{100, 200})},
		{[]int{}, Above_fiftyTransactions([]int{})},
	}
	
	for idx, t := range tests {
		res := Above_fiftyTransactions(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// works i think

File: selfstudy/js/transactions_filterhigh.js
/*
task 35:
Transactions - FilterHigh
You need to return only the values greater than 50 from transaction amounts in dollars
*/

function above_fifty_transactions(amounts) {
  return amounts.filter(x => x > 50);
}

// Tests
const tests = [
  { input: [10, 20, 30], expected: above_fifty_transactions([10, 20, 30]) },
  { input: [50, 60, 70], expected: above_fifty_transactions([50, 60, 70]) },
  { input: [1, 2, 3], expected: above_fifty_transactions([1, 2, 3]) },
  { input: [100, 200], expected: above_fifty_transactions([100, 200]) },
  { input: [], expected: above_fifty_transactions([]) }
];

if (process.argv.includes('mocha')) {
  const assert = require('assert');
  describe('above_fifty_transactions', () => {
    tests.forEach((t, idx) => {
      it(`test case ${idx+1}`, () => {
        assert.deepStrictEqual(above_fifty_transactions(t.input), t.expected);
      });
    });
  });
} else {
  tests.forEach((t, idx) => {
    console.log(`Test ${idx+1}:`, above_fifty_transactions(t.input) === t.expected || JSON.stringify(above_fifty_transactions(t.input)) === JSON.stringify(t.expected) ? "PASS" : "FAIL");
  });
}

// works i think
